### PR TITLE
Fix registration errors

### PR DESCRIPTION
**Pull request by riorio805**
This PR fixes the following issues:
- Fix get the wrong birthday when login google
- Allow manual birthday input when login google

### CODE REVIEW

File: src/main/java/com/safetypin/authentication/service/GoogleAuthService.java [hunk 1]
```diff
@@ -322,13 +329,29 @@ LocalDate extractBirthday(String jsonResponse) {
...
+                    JsonObject source = metadata.getAsJsonObject("source");
+                    if ("ACCOUNT".equals(source.get("type").getAsString())) {
+                        return parseDate(birthdayObj);
+                    }
+                }
+            }
+        }
+
+        // Fallback to the first birthday if "ACCOUNT" type is not found
+        return parseDate(birthdaysArray.get(0).getAsJsonObject());
```
[the remark this answers is not in the document]
Comment (riorio805) [reply src/main/java/com/safetypin/authentication/service/GoogleAuthService.java:346]: Already checked before the for loop.

**Comment by KronosDP** [comment]
Thanks for getting this PR up! Looks like some really solid work going into making our Google auth flow more robust, especially around handling birthdate data and improving error responses.

**Good Stuff:**

* Really appreciate the improved error handling in the controller – mapping that `IllegalArgumentException` to a `BAD_REQUEST` response is a great move for giving clearer feedback to clients. Good practice there!
* Big 👍 on the increased test coverage, particularly for the service logic covering those fallback scenarios and edge cases with missing API data. Testing those paths thoroughly is super important.
* Handling the potential for missing data from the Google API gracefully (by trying to use a backup or throwing an exception) is definitely a step in the right direction for robustness.

**Points to Discuss / Potential Areas:**

* One thing that caught my eye is the addition of the `optional birthdate` field in the `GoogleAuthDTO` for clients to provide a backup. This feels a little unusual for a standard Google OAuth flow where we expect Google to be the source of truth. Could we chat about the specific use case driving this? It might verge on a slightly less ideal practice if it allows clients to potentially provide data that contradicts or bypasses the primary Google source without a very clear reason.
* Also, a quick thought on using `IllegalArgumentException` in the service when a birthdate isn't found from *either* source. While the controller handles it well, sometimes `IllegalArgumentException` is used more specifically for invalid *input arguments* to a method, rather than a failure to acquire necessary *external* data. Would a more specific custom exception (`MissingBirthdateException`, etc.) or perhaps an `IllegalStateException` be a clearer semantic fit here? Not a major issue since it's caught, but worth a quick discussion on our exception strategy.

Overall, good progress! Let's touch base on those couple of points.
